lib/addons/jupyterhub/index.ts

```diff
@@ -173,15 +180,31 @@ export class JupyterHubAddOn extends HelmAddOn {
         const enableIngress = this.options.enableIngress || false;
         const ingressHosts = this.options.ingressHosts || [];
         const ingressAnnotations = this.options.ingressAnnotations;
+        const cert = this.options.certificateResourceName;
         setPath(values, "ingress.enabled", enableIngress);
 
         if (enableIngress){
-            setPath(values, "ingress.annotations", ingressAnnotations);
+            const presetAnnotations: any = {
+                'alb.ingress.kubernetes.io/scheme': 'internet-facing',
+                'alb.ingress.kubernetes.io/target-type': 'ip',
+                'kubernetes.io/ingress.class': 'alb',
+            };
+
+            if (cert){
+                presetAnnotations['alb.ingress.kubernetes.io/ssl-redirect'] = '443';
+                presetAnnotations['alb.ingress.kubernetes.io/listen-ports'] = '[{"HTTP": 80},{"HTTPS":443}]';
+                const certificate = clusterInfo.getResource<ICertificate>(cert);
+                presetAnnotations['alb.ingress.kubernetes.io/certificate-arn'] = certificate?.certificateArn;
+            } 
+
+            const annotations = { ...ingressAnnotations, ...presetAnnotations}
+            setPath(values, "ingress.annotations", annotations);
             setPath(values, "ingress.hosts", ingressHosts);
             setPath(values, "proxy.service", {"type" : "NodePort"});
         } else {
             assert(!ingressHosts || ingressHosts.length == 0, 'Ingress Hosts CANNOT be assigned when ingress is disabled');
             assert(!ingressAnnotations, 'Ingress annotations CANNOT be assigned when ingress is disabled');
+            assert(!cert, 'You cannot provide a certificate for NLB.');
```

Comment:
a cert for NLB? it may be confusing since it does not look like customers specify that they need NLB, you just default to it. I would reword to state "cert option is only support if ingress is enabled" or something similar.

Reply:
addressed